packages/modules/currency/src/services/currency-module-service.ts

```diff
@@ -15,20 +15,15 @@ import { ModulesSdkUtils } from "@medusajs/utils"
 import { Currency } from "@models"
 import { entityNameToLinkableKeysMap, joinerConfig } from "../joiner-config"
 
-const generateMethodForModels = {}
-
 type InjectedDependencies = {
   baseRepository: DAL.RepositoryService
   currencyService: ModulesSdkTypes.IMedusaInternalService<any>
 }
 
 export default class CurrencyModuleService<TEntity extends Currency = Currency>
```

Comment:
Do we need the TEntity extends Currency = Currency now?

Reply:
That is part of the class, not the base class. But I believe we did that in case the user change the Repository layer, in that case he could specify something different here and be applied to the entire class. But with the DML comming I believe we could think about removing it. Lets maybe tackle this one separately as it can have many impacts on some modules more complex